src/components/SingleArticle.jsx

```diff
@@ -16,6 +20,25 @@ const SingleArticle = () => {
         });
       }, [article_id]);
 
+      const handleVote = (voteType) => {
+        if (hasVoted && voteType === article.voteType) {
+          setVoteError("Oooops! You have already voted once!");
+          return;
+        } else if (hasVoted && voteType !== article.voteType) {
+          const newVotesCount = voteType === 'up' ? article.votes + 2 : article.votes - 2;
+          setArticle({ ...article, votes: newVotesCount, voteType });
```

Comment:
The user experience is a little bit weird and I think it's coming from this area of the handleVote function

When I vote one way but change my mind and want to vote the other way it then changes by two votes. It should only change by one vote per button click. 

e.g. for a article with 0 votes
- I upvote, the vote goes up by one (1 vote)
- I upvote again, I'm not allowed to vote (1 vote)
- I downvote, I decrease the votes down by one (0 votes)
- I downvote again, I decrease the votes down by one (-1 votes) because I'm allowed one vote either direction
- I downvote again, I am not allowed to vote (-1 votes)

Please let me know if this isn't clear what I'm asking for here